src/editor/clipboard.js

```diff
@@ -33,12 +34,27 @@ export default class Clipboard {
     item.getAsString((text) => {
       if (isUrl(text) && this.contents.hasSelectedText()) {
         this.contents.createLinkWithSelectedText(text)
+      } else if (isUrl(text)) {
+        const nodeKey = this.contents.createLink(text)
+        this.#dispatchLinkCreatedEvent(nodeKey, { url: text })
       } else {
         this.#pasteMarkdown(text)
       }
     })
   }
 
+  #dispatchLinkCreatedEvent(nodeKey, payload) {
```

Comment:
It'd probably be worth changing it here too for consistency:

```suggestion
  #dispatchLinkInsertEvent(nodeKey, payload) {
```